lms/djangoapps/badges/backends/badgr.py

```diff
@@ -0,0 +1,171 @@
+
+    @lazy
+    def _badge_create_url(self):
+        """
+        URL for generating a new Badge specification
+        """
+        return "{}/badges".format(self._base_url)
+
+    def _badge_url(self, slug):
+        """
+        Get the URL for a course's badge in a given mode.
+        """
+        return "{}/{}".format(self._badge_create_url, slug)
+
+    def _assertion_url(self, slug):
+        """
+        URL for generating a new assertion.
+        """
+        return "{}/assertions".format(self._badge_url(slug))
+
+    def _slugify(self, badge_class):
+        """
+        Get a compatible badge slug from the specification.
+        """
+        slug = badge_class.issuing_component + badge_class.slug
+        if badge_class.issuing_component and badge_class.course_id:
+            # Make this unique to the course, and down to 64 characters.
+            # We don't do this to badges without issuing_component set for backwards compatibility.
+            slug = hashlib.sha256(slug + unicode(badge_class.course_id)).hexdigest()
+        if len(slug) > MAX_SLUG_LENGTH:
+            # Will be 64 characters.
+            slug = hashlib.sha256(slug).hexdigest()
+        return slug
+
+    def _log_if_raised(self, response, data):
+        """
+        Log server response if there was an error.
+        """
+        try:
+            response.raise_for_status()
+        except HTTPError:
+            LOGGER.error(
+                u"Encountered an error when contacting the Badgr-Server. Request sent to %s with headers %s.\n"
+                u"and data values %s\n"
+                u"Response status was %s.\n%s",
+                repr(response.request.url), repr(response.request.headers),
+                repr(data),
+                response.status_code, response.content
+            )
+            raise
+
+    def _create_badge(self, badge_class):
+        """
+        Create the badge class on Badgr.
+        """
+        image = badge_class.image
+        # We don't want to bother validating the file any further than making sure we can detect its MIME type,
+        # for HTTP. The Badgr-Server should tell us if there's anything in particular wrong with it.
+        content_type, __ = mimetypes.guess_type(image.name)
+        if not content_type:
+            raise ValueError(
+                "Could not determine content-type of image! Make sure it is a properly named .png file."
+            )
+        files = {'image': (image.name, image, content_type)}
+        data = {
+            'name': badge_class.display_name,
+            'criteria': badge_class.criteria,
+            'slug': self._slugify(badge_class),
+            'description': badge_class.description,
+        }
+        result = requests.post(self._badge_create_url, headers=self._get_headers(), data=data, files=files)
+        self._log_if_raised(result, data)
+
+    def send_assertion_created_event(self, user, assertion):
```

Comment:
This method is only used in this class.  I'm usually not a fan of obsessive underscore naming, but since you are using `_underscore` names in this class, shouldn't this be one of them?